library/data/registry_entry_attachments/src/main/java/org/quiltmc/qsl/registry/attachment/impl/RegistryEntryAttachmentHolder.java

```diff
@@ -93,21 +94,45 @@ public static <R> RegistryEntryAttachmentHolder<R> getAssets(Registry<R> registr
 	}
 
 	public final Table<RegistryEntryAttachment<R, ?>, R, Object> valueTable;
+	public final Table<RegistryEntryAttachment<R, ?>, TagKey<R>, Object> valueTagTable;
 
 	@SuppressWarnings("UnstableApiUsage")
 	private RegistryEntryAttachmentHolder() {
 		this.valueTable = Tables.newCustomTable(new Object2ReferenceOpenHashMap<>(), Reference2ObjectOpenHashMap::new);
+		this.valueTagTable = Tables.newCustomTable(new Object2ReferenceOpenHashMap<>(), Reference2ObjectOpenHashMap::new);
 	}
 
 	@SuppressWarnings("unchecked")
 	public <V> V getValue(RegistryEntryAttachment<R, V> attachment, R entry) {
-		return (V) this.valueTable.get(attachment, entry);
+		V value = (V) this.valueTable.get(attachment, entry);
+		if (value == null) {
+			Map<TagKey<R>, Object> row = valueTagTable.row(attachment);
+			for (Map.Entry<TagKey<R>, Object> tagValue : row.entrySet()) {
+				for (Holder<R> holder : attachment.registry().getTagOrEmpty(tagValue.getKey())) {
+					if (holder.value().equals(entry)) {
+						if (value != null) {
+							Initializer.LOGGER.warn("Entry {} for Registry {} already has Attachment {} defined. Overriding with value from tag {}",
+									attachment.registry().getId(entry),
+									attachment.registry().getKey().getValue(),
+									attachment.id(),
+									tagValue.getKey().id());
+						}
+						value = (V) tagValue.getValue();
+					}
+				}
+			}
+		}
+		return value;
 	}
 
 	public <T> void putValue(RegistryEntryAttachment<R, T> attachment, R entry, T value) {
 		this.valueTable.put(attachment, entry, value);
 	}
 
+	public <T> void putValueTag(RegistryEntryAttachment<R, T> attachment, TagKey<R> tag, T value) {
```

Comment:
Is naming this method differently really needed?
Could also be named `putValue`.

or `putTaggedValue`
I find `putValueTag` really weird.